From: Thomas Gleixner <tglx@linutronix.de>
Date: Sat, 25 Jul 2009 22:06:27 +0200
Subject: mm: Remove preempt count from pagefault disable/enable

Now that all users are cleaned up, we can remove the preemption count.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 mm/memory.c |    7 -------
 1 file changed, 7 deletions(-)

Index: linux-3.2/mm/memory.c
===================================================================
--- linux-3.2.orig/mm/memory.c
+++ linux-3.2/mm/memory.c
@@ -3439,7 +3439,6 @@ unlock:
 #ifdef CONFIG_PREEMPT_RT_FULL
 void pagefault_disable(void)
 {
-	inc_preempt_count();
 	current->pagefault_disabled++;
 	/*
 	 * make sure to have issued the store before a pagefault
@@ -3457,12 +3456,6 @@ void pagefault_enable(void)
 	 */
 	barrier();
 	current->pagefault_disabled--;
-	dec_preempt_count();
-	/*
-	 * make sure we do..
-	 */
-	barrier();
-	preempt_check_resched();
 }
 EXPORT_SYMBOL_GPL(pagefault_enable);
 #endif
